music21/expressions.py

```diff
@@ -517,21 +519,205 @@ def fillListOfRealizedNotes(
+                        cautionaryNotImmediateRepeat=cautionaryNotImmediateRepeat,
+                        lastNoteWasTied=lastNoteWasTied)
+
+    @property
+    def otherPitches(self) -> tuple[pitch.Pitch, ...]:
+        return self._otherPitches
+
 
 # ------------------------------------------------------------------------------
 class GeneralMordent(Ornament):
     '''
     Base class for all Mordent types.
     '''
-    def __init__(self, **keywords):
+    def __init__(self, *, accid: str | None = None, **keywords):
         super().__init__(**keywords)
-        self.direction = ''  # up or down
-        self.size = None  # interval.Interval (General, etc.) class
+        self._direction = ''  # up or down
+        self._accid: str | None = None
+        if accid is not None and pitch.isValidAccidentalName(accid):
+            self._accid = pitch.makeAccidentalName(accid)
+        self._size: interval.IntervalBase | None = None
         self.quarterLength = 0.125  # 32nd note default
-        self.size = interval.GenericInterval(2)
         self.placement = 'above'
 
-    def realize(self, srcObj: 'music21.note.Note', *, inPlace=False):
+    @property
+    def name(self) -> str:
+        '''
+        returns the name of the Mordent/InvertedMordent, which is generally
+        the class name lowercased, with spaces where a new capital occurs. The
+        name also will include the accidental, if it exists.
+
+        Subclasses can override this as necessary.
+
+        >>> mordent = expressions.Mordent()
+        >>> mordent.name
+        'mordent'
+
+        >>> invertedMordent = expressions.InvertedMordent(accid='sharp')
+        >>> invertedMordent.name
+        'inverted mordent (sharp)'
+
+        '''
+        theName: str = super().name
+        if self.accid is not None:
+            theName += ' (' + self.accid + ')'
+        return theName
+
+    @property
+    def accid(self) -> str | None:
+        return self._accid
+
+    @accid.setter
+    def accid(self, newAccid: str | None):
+        if self._size is not None:
+            raise ExpressionException('Cannot set mordent accid if size already set.')
+
+        # silently ignore invalid accid
+        if newAccid is None:
+            self._accid = None
+        elif pitch.isValidAccidentalName(newAccid):
+            self._accid = pitch.makeAccidentalName(newAccid)
+
+    @property
+    def size(self) -> interval.IntervalBase | None:
+        if self._size is None:
+            raise ExpressionException(
+                'Cannot compute mordent size without srcPitch unless .size has been '
+                + 'explicitly set (use mordent.getSize(srcPitch) instead.)'
+            )
+        return self._size
+
+    @size.setter
+    def size(self, newSize: interval.IntervalBase | None):
+        if self._accid is not None:
+            raise ExpressionException('Cannot set mordent size if accid already set.')
+        self._size = newSize
```

Comment:
only raise exceptions in setters if absolutely needed.  Have the setter change the accidentalName to '' if needed.

Reply:
Done.